src/main/java/org/apache/commons/lang3/text/StrSubstitutor.java

```diff
@@ -533,81 +744,136 @@ protected boolean substitute(StrBuilder buf, int offset, int length) {
      * @return the length change that occurs, unless priorVariables is null when the int
      *  represents a boolean flag as to whether any change occurred.
      */
-    private int substitute(StrBuilder buf, int offset, int length, List priorVariables) {
-        StrMatcher prefixMatcher = getVariablePrefixMatcher();
-        StrMatcher suffixMatcher = getVariableSuffixMatcher();
-        char escape = getEscapeChar();
-        
-        boolean top = (priorVariables == null);
+    private int substitute(final StrBuilder buf, final int offset, final int length, List<String> priorVariables) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 22. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.